### PR TITLE
Do-Not-Track opt-in setting

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Adds logic to scout.js to read Do-Not-Track opt-ing form the server settings query object and user preference form the window and decide whether to load the script. This will allow sites embedding the script to configure whether they want the user DNT preference to be enforced.

### Why?
We want to be good internet citizens. Whilst the insights.js script does not collect personally identifiable data, we recognise that some people enable this setting in an effort to stop sites collecting any data at all. This has come up in discussions with early beta users of the script.

Therefore, to ensure we respect the intentions of end-users and users of the script we are allowing the functionality to be configurable via a query param`?dnt=true` which gets reflected in the settings object. 

We currently do not run the script if DNT is enabled, however after this change we will default to always collecting and embedding websites can disable the functionality via this query param if they wish. 

### How?
It reads the setting configured from by the host website form query parameter and the user preference from the `doNotTrack` api and prevents the library from running if both are true.

### CODE REVIEW

**Comment by JakeChampion** [comment]
@phamann Does the fact that this pull-request being closed mean that insights.js will not support the DNT header?
